File: core/src/main/java/dev/bee/kanjianki/core/AutoSyncSettingsTogglePolicy.java
package dev.bee.kanjianki.core;

public final class AutoSyncSettingsTogglePolicy {
    public static final String ENABLED_MESSAGE = "Daily Anki sync turned on.";
    public static final String DISABLED_MESSAGE = "Daily Anki sync turned off.";

    private AutoSyncSettingsTogglePolicy() {
    }

    public static ToggleResult enable() {
        return new ToggleResult(true, ENABLED_MESSAGE);
    }

    public static ToggleResult disable() {
        return new ToggleResult(false, DISABLED_MESSAGE);
    }

    public record ToggleResult(boolean enabled, String message) {
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/AutoSyncSettingsTogglePolicy.kt
package dev.bee.kanjianki.core

import java.util.Locale

object AutoSyncSettingsTogglePolicy {
    const val ENABLED_MESSAGE: String = "Daily sync turned on."
    const val DISABLED_MESSAGE: String = "Daily sync turned off."

    private const val JAPANESE_LANGUAGE = "ja"

    @JvmStatic
    fun enable(): ToggleResult {
        return ToggleResult(true, localizedText(ENABLED_MESSAGE, "毎日の同期をオンにしました。"))
    }

    @JvmStatic
    fun disable(): ToggleResult {
        return ToggleResult(false, localizedText(DISABLED_MESSAGE, "毎日の同期をオフにしました。"))
    }

    private fun localizedText(english: String, japanese: String): String =
        if (isJapaneseLocale()) japanese else english

    private fun isJapaneseLocale(): Boolean = Locale.getDefault().language == JAPANESE_LANGUAGE

    @JvmRecord
    data class ToggleResult(val enabled: Boolean, val message: String) {
        override fun toString(): String {
            return "ToggleResult[enabled=$enabled, message=$message]"
        }
    }
}
